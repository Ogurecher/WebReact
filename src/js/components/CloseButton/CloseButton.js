import React, { Component } from 'react';
import { removeFromFavourites } from '../../actions/index';
import { connect } from 'react-redux';

export class CloseButton extends Component {
    constructor(props) {
        super(props);

        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit(event) {
        event.preventDefault();
        
        this.props.removeFromFavourites(this.props.city.position.city);
    }
    
    render() {
        return (
            <form onSubmit={this.handleSubmit}>
                <input type='submit' value='X' />
            </form>
        );
    }
}

function mapStateToProps(state) {
    return {};
}

export default connect(
    mapStateToProps,
    { removeFromFavourites }
)(CloseButton);