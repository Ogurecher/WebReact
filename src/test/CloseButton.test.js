import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import CloseButton from '../js/components/CloseButton/CloseButton';

const store = configureMockStore();

describe('CloseButton', function() {
    test('snapshot renders', function() {
        const cityStub = {
            position: {
                city: 'London'
            }
        };

        const storeMock = store();

        const component = renderer.create(<Provider store={storeMock} ><CloseButton city={cityStub} /></Provider>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});