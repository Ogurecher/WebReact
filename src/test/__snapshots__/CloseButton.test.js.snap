// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton snapshot renders 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    type="submit"
    value="X"
  />
</form>
`;
